spec/unit/pops/serialization/packer_spec.rb

```diff
@@ -0,0 +1,151 @@
+require 'spec_helper'
+require 'puppet/pops/serialization'
+
+module Puppet::Pops::Serialization
+[JSON].each do |packer_module|
+describe "the Puppet::Pops::Serialization when using #{packer_module.name}" do
+  let(:io) { StringIO.new }
+  let(:reader_class) { packer_module::Reader }
+  let(:writer_class) { packer_module::Writer }
+
+  def write(*values)
+    io.reopen
+    serializer = writer_class.new(io)
+    values.each { |val| serializer.write(val) }
+    serializer.finish
+    io.rewind
+  end
+
+  def read(count = nil)
+    @deserializer = reader_class.new(io)
+    count.nil? ? @deserializer.read : Array.new(count) { @deserializer.read }
+  end
+
+  context 'can write and read a Scalar' do
+    it 'String' do
+      val = 'the value'
+      write(val)
+      val2 = read
+      expect(val2).to be_a(String)
+      expect(val2).to eql(val)
+    end
+
+    it 'Integer' do
+      val = 32
+      write(val)
+      val2 = read
+      expect(val2).to be_a(Integer)
+      expect(val2).to eql(val)
+    end
+
```

Comment:
See earlier comment on cap to 64 bit values I think this should be enforced or there is risk for silent bad behavior.